language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/rename/RenameUtil.java

```diff
@@ -51,8 +68,26 @@ private RenameUtil() {
             throw new UserErrorException("Invalid identifier provided");
         }
 
-        Map<Module, List<Location>> locationMap = ReferencesUtil.getReferences(context);
+        fillTokenInfoAtCursor(context);
+
+        Optional<Document> document = context.currentDocument();
+        if (document.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        Range cursorPosRange = new Range(context.getCursorPosition(), context.getCursorPosition());
+        NonTerminalNode nodeAtCursor = CommonUtil.findNode(cursorPosRange, document.get().syntaxTree());
+
+        if (onQNameReferenceNode(context, nodeAtCursor)) {
```

Comment:
Can't we use the `QNameReferencesUtil` here? It has a util called `onQualifiedNameIdentifier`. We can use modify it to accept a `PositionedOperationContext`